pkg/controller/controller.go

```diff
@@ -1298,24 +1315,58 @@ func (s *Service) ControllerExpandVolume(ctx context.Context, req *csi.Controlle
 		return nil, status.Errorf(codes.OutOfRange, "volume exceeds allowed limit")
 	}
 
+	powerStoreClient := s.Arrays()[arrayID].Client
+
 	if protocol == "scsi" {
-		vol, err := s.Arrays()[arrayID].Client.GetVolume(ctx, id)
+		vol, err := powerStoreClient.GetVolume(ctx, id)
 		if err != nil {
 			return nil, status.Errorf(codes.OutOfRange, "detected SCSI protocol but wasn't able to fetch the volume info")
 		}
+		isMetroSession := remoteArrayID != ""
+
+		// get session id
+		replicationSessionId := ""
+		if isMetroSession {
+			vgs, err := powerStoreClient.GetVolumeGroupsByVolumeID(ctx, req.GetVolumeId())
+			if err != nil {
+				return nil, err
+			}
+			if len(vgs.VolumeGroup) == 0 {
+				replicationSessionId = vol.MetroReplicationSessionID
+			} else {
+				vg := vgs.VolumeGroup[0]
+				rs, err := powerStoreClient.GetReplicationSessionByLocalResourceID(ctx, vg.ID)
+				if err != nil {
+					return nil, err
+				}
+				replicationSessionId = rs.ID
+			}
+		}
+
 		if vol.Size < requiredBytes {
-			_, err = s.Arrays()[arrayID].Client.ModifyVolume(context.Background(), &gopowerstore.VolumeModify{Size: requiredBytes}, id)
+			_, err = pauseSessionIfMetro(isMetroSession, powerStoreClient, ctx, replicationSessionId)
+			if err != nil {
+				return nil, err
+			}
+
+			_, err = powerStoreClient.ModifyVolume(context.Background(), &gopowerstore.VolumeModify{Size: requiredBytes}, id)
```

Comment:
should probably log this error if any exists, in case we also hit `resumeSessionErr` and don't get a chance to return this error.
Or combine and return both errors.

Reply:
Updated